File: src/tools/semgrep/eval-repos/synthetic/typescript/empty_catch.ts
/**
 * Test file for DD smell D1_EMPTY_CATCH detection.
 * Contains multiple empty catch blocks.
 */

interface User {
    id: number;
    name: string;
    email: string;
}

async function fetchUserData(userId: number): Promise<User | null> {
    try {
        const response = await fetch(`/api/users/${userId}`);
        return await response.json();
    } catch (error) {
        // D1_EMPTY_CATCH: Empty catch block - swallows errors silently
    }
    return null;
}

function parseJsonSafe<T>(jsonString: string): T | null {
    try {
        return JSON.parse(jsonString) as T;
    } catch (e) {
        // D1_EMPTY_CATCH: Another empty catch
    }
    return null;
}

async function processItems<T, R>(items: T[], transform: (item: T) => R): Promise<R[]> {
    const results: R[] = [];
    for (const item of items) {
        try {
            const processed = transform(item);
            results.push(processed);
        } catch (error) {
            // D1_EMPTY_CATCH: Empty catch in loop
        }
    }
    return results;
}

class DataService {
    private cache: Map<string, unknown> = new Map();

    async loadData(key: string): Promise<unknown> {
        try {
            const data = await this.fetchFromApi(key);
            this.cache.set(key, data);
            return data;
        } catch (err) {
            // D1_EMPTY_CATCH: Silently ignore errors
        }
        return null;
    }

    private async fetchFromApi(key: string): Promise<unknown> {
        return {};
    }

    clearCache(): void {
        try {
            this.cache.clear();
        } catch {
            // D1_EMPTY_CATCH: Bare catch with empty body
        }
    }
}

export { User, DataService, fetchUserData, parseJsonSafe, processItems };

File: src/tools/semgrep/eval-repos/synthetic/python/empty_catch.py
"""
Test file for DD smell D1_EMPTY_CATCH detection.
Contains multiple empty catch blocks.
"""

import json


def process_user_data(user_id: int) -> dict:
    """Process user data with problematic exception handling."""
    try:
        # Fetch user from database
        user = fetch_user(user_id)
        return user
    except Exception:
        # D1_EMPTY_CATCH: Empty catch block - swallows all errors silently
        pass


def load_config(path: str) -> dict:
    """Load configuration from file."""
    try:
        with open(path, 'r') as f:
            return json.load(f)
    except FileNotFoundError:
        # D1_EMPTY_CATCH: Another empty catch
        pass
    except json.JSONDecodeError:
        # D1_EMPTY_CATCH: Yet another empty catch
        pass
    return {}


def dangerous_operation(data: list) -> None:
    """Perform operations that might fail."""
    for item in data:
        try:
            process_item(item)
        except ValueError:
            # D1_EMPTY_CATCH: Empty catch in loop
            pass
        except KeyError:
            # D1_EMPTY_CATCH: Multiple empty catches
            pass


def fetch_user(user_id: int) -> dict:
    """Stub function."""
    return {"id": user_id, "name": "Test"}


def process_item(item: any) -> None:
    """Stub function."""
    pass


# Another example with bare except
def risky_calculation(x: int, y: int) -> int:
    """Calculate division with poor error handling."""
    try:
        return x // y
    except:
        # D1_EMPTY_CATCH: Bare except with empty body
        pass
    return 0
